src/screens/Accounts/Accounts.js

```diff
@@ -295,40 +312,114 @@ class AccountsScreen extends React.Component<Props, State> {
+
+      case 'NEW_SMART_WALLET':
+        return this.renderInitializeAccount(item);
+
+      default:
+        return null;
+    }
   };
 
-  render() {
-    const { showPinModal, changingAccount } = this.state;
+  visibleAccounts(accounts: Accounts, smartWalletFeatureEnabled: boolean): Accounts {
+    if (smartWalletFeatureEnabled) {
+      return accounts;
+    }
+
+    return accounts.filter(({ type }) => type !== ACCOUNT_TYPES.SMART_WALLET);
+  }
+
+  wallets(): ListItem[] {
     const {
       accounts,
-      blockchainNetworks,
-      smartWalletFeatureEnabled,
-      smartWalletState,
       navigation,
       balances,
       assets,
       rates,
+      smartWalletFeatureEnabled,
+      blockchainNetworks,
+      baseFiatCurrency,
     } = this.props;
-    const ppnNetwork = blockchainNetworks.find((network) => network.id === BLOCKCHAIN_NETWORK_TYPES.PILLAR_NETWORK);
-    const smartWalletStatus: SmartWalletStatus = getSmartWalletStatus(accounts, smartWalletState);
-    const showSmartWalletInitButton = !smartWalletStatus.hasAccount && smartWalletFeatureEnabled;
-    const visibleAccounts = smartWalletFeatureEnabled
-      ? accounts
-      : accounts.filter(({ type }) => type !== ACCOUNT_TYPES.SMART_WALLET);
-
-    const visibleAccountsWithBalance = visibleAccounts.map((acc) => {
-      const accountBalances = balances[acc.id] || {};
-      const balance = calculatePortfolioBalance(assets, rates, accountBalances);
-      return { ...acc, balance };
+
+    const visibleAccounts = this.visibleAccounts(accounts, smartWalletFeatureEnabled);
+    const hasAccount = hasSmartWalletAccount(accounts);
+    const showSmartWalletInitButton = !hasAccount && smartWalletFeatureEnabled;
+    const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
+
+    const visibleAccountsWithBalance = visibleAccounts.map((account: Account): ListItem => {
+      const accountBalances = balances[account.id];
+      if (accountBalances) {
+        // TODO: improve balance calculation?
+        const balance = calculatePortfolioBalance(assets, rates, accountBalances);
+        const balanceInFiat = balance && balance[fiatCurrency];
+
+        if (balanceInFiat) {
+          return { type: 'ACCOUNT', balanceInFiat, account };
+        }
+      }
+
+      return { type: 'ACCOUNT', account };
     });
 
-    const walletsToShow = showSmartWalletInitButton
+    const wallets = showSmartWalletInitButton
       ? [
         ...visibleAccountsWithBalance,
-        { type: 'SMART_WALLET_INIT', action: () => navigation.navigate(SMART_WALLET_INTRO) },
+        { type: 'NEW_SMART_WALLET', action: () => navigation.navigate(SMART_WALLET_INTRO) },
       ]
       : visibleAccountsWithBalance;
 
+    const ppnNetwork = blockchainNetworks.find((network) => network.id === BLOCKCHAIN_NETWORK_TYPES.PILLAR_NETWORK);
+
+    if (smartWalletFeatureEnabled && ppnNetwork) {
+      return [
+        ...wallets,
+        { type: 'NETWORK', network: ppnNetwork },
```

Comment:
Hm, why we need ppnNetwork listed in this.wallets? 
And then included once again in line 443?
If we are keeping ppnNetwork in this.wallets, maybe lets rename the prop to accounts?